lib/ansible/modules/windows/win_audit_rule.ps1

```diff
@@ -0,0 +1,182 @@
+    ($rights -split ',').Trim() | % {
+        if ($_ -notin $PossibleRights)
+        {
+            Fail-Json -obj $result -message "$_ does not seem to be a valid FILE SYSTEM right"
+        }
+    }
+    $FileSystem = $True
+}
+
+$ACL = Get-Acl $path -Audit
+
+#configure acl object to remove the specified user
+If ($state -eq 'absent')
+{
+    #Try and find an identity on the object that matches identity_reference
+    $ToRemove = ($ACL.Audit | ? {$_.IdentityReference.Translate([System.Security.Principal.SecurityIdentifier]) -eq $SID}).IdentityReference
+
+    #Exit with changed false if no identity is found
+    If (-Not $ToRemove)
+    {
+        $result.current_audit_rules = Get-CurrentAuditRules $path
+        Exit-Json -obj $result -message
+    }
+
+    #update the ACL object if identity found
+    Try
+    {
+        $ACL.PurgeAuditRules($ToRemove)
+    }
+    Catch
+    {
+        $result.current_audit_rules = Get-CurrentAuditRules $path
+        fail-Json -obj $result -message "Failed to remove auditrule: $($_.Exception.Message)"
+    }
+
+}
+#otherwise configure acl object to add/modify user
+Else
+{
+    $flags = [System.Security.AccessControl.AuditFlags]$audit_flags
+    $inherit = [System.Security.AccessControl.InheritanceFlags]$inheritance_flags
+    $prop = [System.Security.AccessControl.PropagationFlags]$propagation_flags
+
+    If ($FileSystem)
+    {
+        $rights = [System.Security.AccessControl.FileSystemRights]$rights
+        $AccessRule = New-Object System.Security.AccessControl.FileSystemAuditRule($identity_reference,$rights,$inheritance_flags,$propagation_flags,$audit_flags)
+    }
+    Else
+    {
+        $rights = [System.Security.AccessControl.RegistryRights]$rights
+        $AccessRule = New-Object System.Security.AccessControl.RegistryAuditRule($identity_reference,$rights,$inheritance_flags,$propagation_flags,$audit_flags)
+    }
+
+    Foreach ($group in $ACL.Audit)
+    {
+        #exit here if any existing rule matches defined rule since no change is needed
+        If (
+            ($group | select -expand "*Rights") -eq $rights -and
+            $group.AuditFlags -eq $flags -and
+            $group.IdentityReference.Translate([System.Security.Principal.SecurityIdentifier]) -eq $SID -and
+            $group.InheritanceFlags -eq $inherit -and
+            $group.PropagationFlags -eq $prop
+        )
+        {
+            $result.current_audit_rules = Get-CurrentAuditRules $path
+            Exit-Json -obj $result -message
+        }
+    }
+
+
+    Try
+    {
+        $ACL.SetAuditRule($AccessRule)
+    }
+    Catch
+    {
+        fail-Json -obj $result -message "Failed to set the audit rule: $($_.Exception.Message)"
+    }
+}
+
+#finally set the permissions
+Try {
+    Set-Acl -Path $path -ACLObject $ACL -WhatIf:$check_mode
+}
+Catch {
+    $result.current_audit_rules = Get-CurrentAuditRules $path
+    Fail-Json -message "Failed to apply audit change: $($_.Exception.Message)" $result
```

Comment:
Are you able to keep this similar to the other calls, e.g. `Fail-Json -obj $result -message ""`